.github/workflows/check.yml

```
@@ -7,11 +7,19 @@ on:
       - main
   # ..and any pull request.
   pull_request:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'The branch, tag or SHA to checkout (uses the default branch if empty)'
+  # Cron job to run checks @ 8:30 pm daily on the latest commit on the default branch - main
+  schedule:
+    - cron: '30 20 * * *'
 
 # Cancel any in progress run of the workflow for a given PR
 # This avoids building outdated code
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref }}
+  # Fallback used github.head_ref as it only defined on pull_request
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

I propose we do this:
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
```

This way it only runs 1 instance for a given commit when triggered manually (without interfering with other branches).
We should do that for all workflows.